src/processors/default/createERC721NFTsFromTransfersProcessor.ts

```diff
@@ -31,38 +32,60 @@ const processorFunction = (contracts: NftFactory[]) =>
+      const nftId = contractType.buildNFTId(contractAddress, tokenId);
+
       transfers.push({
-        id: `${CHAIN}/${item.blockNumber}/${item.logIndex}`,
-        contractAddress: formatAddress(contract.id),
-        from: item.args!.from,
-        to: item.args!.to,
+        id: ethereumTransferId(item.blockNumber, item.logIndex),
+        contractAddress: contractAddress,
+        from: fromAddress,
+        to: toAddress,
         tokenId,
         createdAtEthereumBlockNumber: '' + item.blockNumber,
-        nftId: contractType.buildNFTId(contract.id, tokenId), 
+        nftId: nftId,
         transactionHash: item.transactionHash
       });
-      if (!newMint) {
-        updates.push({
-          id: contractType.buildNFTId(contract.id, tokenId),
-          owner: item.args!.to
+
+      if (!newMint(fromAddress)) {
+        updatedNFTs.push({
+          id: nftId,
+          owner: toAddress,
+          burned: burned(toAddress)
         })
-        return undefined;
+        nftsCollectorsChanges.push({ nftId: nftId, collectorId: fromAddress, amount: -1 })
+        if (!burned(toAddress)) {
+          nftsCollectorsChanges.push({ nftId: nftId, collectorId: toAddress, amount: 1 })
+        }
+      } else {
+        newNFTs.push({
+          id: nftId,
+          createdAtEthereumBlockNumber: '' + item.blockNumber,
+          contractAddress: contractAddress,
+          tokenId,
+          platformId: contract.platformId,
+          owner: toAddress,
+          approved: contract.autoApprove
+        });
+        nftsCollectorsChanges.push({ nftId: nftId, collectorId: toAddress, amount: 1 })
       }
-      newNFTs.push({
-        id: contractType.buildNFTId(contract.id, tokenId),
-        createdAtEthereumBlockNumber: '' + item.blockNumber,
-        contractAddress: formatAddress(contract.id),
-        tokenId,
-        platformId: contract.platformId,
-        owner: item.args!.to,
-        approved: contract.autoApprove
-      });
     });
 
+    const allCollectors = nftsCollectorsChanges.map<Collector>(({ collectorId }) => { return { id: collectorId } } );
+    const idPairs = _.uniq(nftsCollectorsChanges.map( (e) => [e.nftId, e.collectorId] ))
+    const existingNftsCollectors = await clients.db.getRecords<NFTsCollectors>(Table.nftsCollectors,
+      [ ['whereIn', [ ['nftId', 'collectorId'], idPairs ] ] ]
+    );
+
+    const allNftsCollectorsChanges = nftsCollectorsChanges.concat(existingNftsCollectors)
+    const updatedNftsCollectors = consolidate(allNftsCollectorsChanges)
+
     await clients.db.insert(Table.nfts, newNFTs.filter(n => !!n), { ignoreConflict: 'id' });
-    await clients.db.update(Table.nfts, updates);
+    await clients.db.upsert(Table.nfts, updatedNFTs);
+    await clients.db.insert(Table.collectors, _.uniqBy(allCollectors, 'id'), { ignoreConflict: 'id' })
+    await clients.db.insert(Table.nftsCollectors, updatedNftsCollectors, { ignoreConflict: ['nftId', 'collectorId'] });
```

Comment:
these are both basically upserts right? should we just use our upsert function if it is capable of doing the same thing?

Reply:
Yep, should be the same result. I'll switch to `upsert`